tracing/tracer.py

```diff
@@ -101,7 +101,7 @@ def _on_trace_is_called(self, frame, event, arg: any) -> typing.Callable:
             # Note: The value error does not stop the program for some reason.
             raise ValueError("The event" + str(event) + " is unknown.")
 
-        if names2types:
+        if names2types and category:
```

Comment:
Raising an error if the category is 0 might be better.

Reply:
This check is done in order to appease the mypy checker. Otherwise there is a warning that category may be None.

Reply:
Also, raising Exceptions within the tracer is useless, as per the Python documentation:
`If there is any error occurred in the trace function, it will be unset, just like settrace(None) is called.`